src/client/display.c

```diff
@@ -675,6 +714,54 @@ display_interfaces(lldpctl_conn_t *conn, struct writer *w,
 	tag_end(w);
 }
 
+
+/**
+ * Display information about local interfaces.
+ *
+ * @param conn       Connection to lldpd.
+ * @param w          Writer.
+ * @param hidden     Whatever to show hidden ports.
+ * @param env        Environment from which we may find the list of ports.
+ * @param details    Level of details we need (DISPLAY_*).
+ */
+void
+display_local_interfaces(lldpctl_conn_t *conn, struct writer *w,
+    struct cmd_env *env,
+    int hidden, int details)
+{
+	lldpctl_atom_t *iface;
+	int protocol = LLDPD_MODE_MAX;
+	const char *proto_str;
+
+	/* user might have specified protocol to filter display results */
+	proto_str = cmdenv_get(env, "protocol");
+
+	if (proto_str) {
+		log_debug("display", "filter protocol: %s ", proto_str);
+
+		protocol = 0;
+		for (lldpctl_map_t *protocol_map =
+			 lldpctl_key_get_map(lldpctl_k_port_protocol);
+		     protocol_map->string;
+		     protocol_map++) {
+			if (!strcasecmp(proto_str, protocol_map->string)) {
+				protocol = protocol_map->value;
+				break;
+			}
+		}
+	}
```

Comment:
This code seems to be unreachable: there is no way to specify a protocol for a local interface.

Reply:
Right.  This can be removed, as if I understand correctly, the protocol used by a local interface is determined by what it receives.